### PR TITLE
Extending NetworkService API #531

### PR DESCRIPTION
**Pull request by nickolaev**

### CODE REVIEW

File: controlplane/pkg/apis/registry/gen.go [hunk 1]
```diff
@@ -1,3 +1,3 @@
 package registry
 
-//go:generate protoc -I . registry.proto --go_out=plugins=grpc:. --proto_path=$GOPATH/src
+//go:generate protoc -I ../../../../vendor -I . registry.proto --go_out=plugins=grpc:. --proto_path=$GOPATH/src
```
Comment (nickolaev) [inline controlplane/pkg/apis/registry/gen.go]: I don't like this approach but that's something working. Could there be a better way to point protocol to the `vendor` folder?

Comment (edwarnicke) [reply controlplane/pkg/apis/registry/gen.go]: So interestingly, I don't have to do this when I use vendored things:
https://github.com/ligato/networkservicemesh/blob/master/controlplane/pkg/apis/local/connection/connection.proto#L23

https://github.com/ligato/networkservicemesh/blob/master/controlplane/pkg/apis/local/connection/gen.go#L3

What problem did you hit that led to this?

File: controlplane/pkg/apis/registry/registry.proto [hunk 2]
```diff
@@ -17,6 +18,15 @@ message NetworkServiceEndpoint {
 message NetworkService {
     string name = 1;
     string payload = 2;
+    message Match {
+        k8s.io.apimachinery.pkg.apis.meta.v1.LabelSelector source_selector = 1;
+        message Destination{
+            k8s.io.apimachinery.pkg.apis.meta.v1.LabelSelector destination_selector = 1;
+            uint32 weight = 2;
```
Comment (edwarnicke) [inline controlplane/pkg/apis/registry/registry.proto]: What did you have in mind for weight?

File: controlplane/pkg/apis/registry/registry.proto [hunk 3]
```diff
@@ -17,6 +18,15 @@ message NetworkServiceEndpoint {
 message NetworkService {
     string name = 1;
     string payload = 2;
+    message Match {
+        k8s.io.apimachinery.pkg.apis.meta.v1.LabelSelector source_selector = 1;
+        message Destination{
```
Comment (edwarnicke) [inline controlplane/pkg/apis/registry/registry.proto]: Could we define this not-inline in another message?  It turns out to make the generated Go super messy.

File: controlplane/pkg/apis/registry/registry.proto (hunk 2 above)
Comment (edwarnicke) [reply controlplane/pkg/apis/registry/registry.proto]: Oh wait... of course, 'weight'...

File: controlplane/pkg/apis/registry/registry.proto [hunk 4]
```diff
@@ -17,6 +18,15 @@ message NetworkServiceEndpoint {
...
+            k8s.io.apimachinery.pkg.apis.meta.v1.LabelSelector destination_selector = 1;
+            uint32 weight = 2;
+        }
+        Destination route = 2;
```
Comment (edwarnicke) [inline controlplane/pkg/apis/registry/registry.proto]: Should this be

repeated Destination

?

File: k8s/pkg/apis/networkservice/v1/types.go [hunk 5]
```diff
@@ -25,7 +25,18 @@ type NetworkService struct {
...
 
 type NetworkServiceSpec struct {
-	Payload string `json:"payload"`
+	Payload string  `json:"payload"`
+	Matches []Match `json:"matches"`
+}
+
+type Match struct {
+	SourceSelector *metaV1.LabelSelector `json:"sourceSelector,omitempty"`
+	Route          Destination           `json:"route"`
```
Comment (edwarnicke) [inline k8s/pkg/apis/networkservice/v1/types.go]: Should this be a []*Destination ?

File: controlplane/pkg/apis/registry/registry.proto (hunk 4 above)
Comment (nickolaev) [reply controlplane/pkg/apis/registry/registry.proto]: I was not sure about this either.

**Pull request merged**